### PR TITLE
015 edit a role

### PR DESCRIPTION
**Pull request by JennicaMistry**

### CODE REVIEW

File: src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public String editJobRole(int jobRoleID,JobRoleWithoutLink jobRole) throws SQLException {
+        try{
+            DataBaseConnection dataBaseConnection = new DataBaseConnection();
+            Connection myConnection = dataBaseConnection.getConnection();
+            String sql = "update jobRoles " +
+                    "set roleName=? ,specification= ? ," +
+                    " bandID=?, capabilityID=?,responsibility=?" +
+                    " where jobRoleID=?;";
+//                    "(\"" + jobRole.getRoleName() + "\",\"" + jobRole.getSpecification() + "\",\"" + jobRole.getBandID() + "\",\"" + jobRole.getCapabilityID() + "\",\"" + jobRole.getResponsibility() + "\",\"" + jobRoleID + "\");";
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java:22]: Remove commented out code

File: src/test/java/com/kainos/ea/dao/RoleFeatureLevelTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    public void editJobRole_shouldThrowSQLException_whenProvoked() throws SQLException {
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+        Mockito.when(roleFeaturesLevel.editJobRole(1,jobRole)).thenThrow(SQLException.class);
+
+        assertThrows(SQLException.class,
+                () -> roleFeaturesLevel.editJobRole(1,jobRole));
+    }
+}
+
```
Comment (hellofadude) [inline src/test/java/com/kainos/ea/dao/RoleFeatureLevelTest.java:47]: remove extra lines and end of file

File: src/test/java/com/kainos/ea/integration/RoleFeatureIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import javax.ws.rs.core.Response;
+import java.util.List;
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class RoleFeatureIntegrationTest {
+    static final DropwizardAppExtension<WebServiceConfiguration> APP = new DropwizardAppExtension<>(
+            WebServiceApplication.class, null,
+            new ResourceConfigurationSourceProvider()
+    );
+
+
```
Comment (hellofadude) [inline src/test/java/com/kainos/ea/integration/RoleFeatureIntegrationTest.java:25]: remove extra line

File: src/test/java/com/kainos/ea/integration/RoleFeatureIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            int response = APP.client().target("http://localhost:8080/api/editJobRole/58")
+                    .request()
+                    .put(entity).getStatus();
+
+            Assertions.assertEquals(response, 500);
+        }
+        catch(Exception e){
+
+
```
Comment (hellofadude) [inline src/test/java/com/kainos/ea/integration/RoleFeatureIntegrationTest.java:70]: Remove extra lines here and lines 75-76

File: src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            statement.executeUpdate();
+            return "Data updated successfully";
+
+        } catch (SQLException e) {
+            throw e;
+
+        }
+
+    }
+
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java:47]: remove extra lines and make consistent with rest of code

**Review by hellofadude — CHANGES_REQUESTED** [review]
Minor comments

File: src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                    "set roleName=? ,specification= ? ," +
+                    " bandID=?, capabilityID=?,responsibility=?" +
+                    " where jobRoleID=?;";
+
+            System.out.println(sql);
+            System.out.println(jobRole.getRoleName());
+            System.out.println(jobRole.getSpecification());
+            System.out.println(jobRole.getResponsibility());
+            System.out.println(jobRole.getCapabilityID());
+            System.out.println(jobRole.getBandID());
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java:28]: do you need these `System.out.println` statements? I think you can remove them

File: src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            PreparedStatement statement = myConnection.prepareStatement(sql);
+            statement.setString(1, jobRole.getRoleName());
+            statement.setString(2, jobRole.getSpecification());
+            statement.setInt(3, jobRole.getBandID());
+            statement.setInt(4, jobRole.getCapabilityID());
+            statement.setString(5, jobRole.getResponsibility());
+            statement.setInt(6, jobRoleID);
+            System.out.println(statement);
```
Comment (hellofadude) [inline src/main/java/com/kainos/ea/dao/RoleFeaturesLevel.java:36]: Here as well

File: src/test/java/com/kainos/ea/dao/RoleFeatureLevelTest.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        jobRole.setCapabilityID(1);
+        jobRole.setSpecification("As a Data Analyst (Associate) in Kainos, you’ll be responsible for matching the needs of data insight with understanding of the available data. Data analysts work closely with customers to produce insight products including reports, dashboards and visualisations but also contribute to project understanding of existing data structures so that inputs and outputs are fully understood.");
+        jobRole.setRoleName("Data Analyst - edited");
+        jobRole.setResponsibility("You’ll be accountable for successful delivery of contemporary data solutions across multiple customers.");
+        Mockito.when(roleFeaturesLevel.editJobRole(1,jobRole)).thenReturn(expectedResult);
+
+        String result = roleFeaturesLevel.editJobRole(1,jobRole);
+
+        assertEquals("Data updated successfully",result);
+    }
```
Comment (hellofadude) [inline src/test/java/com/kainos/ea/dao/RoleFeatureLevelTest.java:37]: Add extra line here to signify start of new method

**Review by hellofadude — APPROVED** [review]

**Review by marcoatkainos — APPROVED** [review]

**Review by Kahrym — APPROVED** [review]

**Pull request merged**